pkg/v10/resource/chart/current.go

```diff
@@ -12,61 +12,70 @@ import (
 
 // GetCurrentState gets the state of the chart in the guest cluster.
 func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interface{}, error) {
-	r.logger.LogCtx(ctx, "level", "debug", "message", "finding chart-operator chart in the guest cluster")
+	var err error
 
-	tenantHelmClient, err := r.getTenantHelmClient(ctx, obj)
-	if tenantcluster.IsTimeout(err) {
-		r.logger.LogCtx(ctx, "level", "debug", "message", "did not get a Helm client for the guest cluster")
+	var tenantHelmClient helmclient.Interface
+	{
```

Comment:
This client creation section is not aligned similar to other two in this PR. Would it make sense to align or even extract to separate function because from quick look it seems they are identical?

Reply:
I can only guess but the errors handled here are different because `GetCurrentState` is called for creating, updating and deleting. So here we have to deal with the intersection of errors dealt with in other places. @rossf7 is that right or why is the error handling different?

Reply:
Checking for `IsTillerInstallationFailed` is missing here basically.

Reply:
Ack

Reply:
I think it's an oversight and `IsTillerInstallationFailed` should also be checked for.

To calculate the current state we need to connect to Tiller. `EnsureTillerInstalled` is called from `getTenantHelmClient` so we should also check for this error.

Reply:
I think current state is wrong there and we should only cancel the resource in all cases.

https://github.com/giantswarm/cluster-operator/blob/38d950d49854c1df26c9e4091d80a6885919ab87/service/controller/aws/v10/resource_set.go#L293-L296

The configmap and chartconfig resources just need a connection to the API server and not Tiller. So we can apply changes there and chart-operator will reconcile them when it can connect to Tiller.

Reply:
Alright. Sorted.